### DISCUSSION TITLE
【要望】DPIの一時的な変更機能

### DISCUSSION THREAD

**Discussion by garbx86** [Yowkees/keyball#340] [Ideas]
Keyball61をWQHD解像度xデュアルディスプレイという環境で使用していますが、5120x1440という広いデスクトップ上での移動と時々必要になる精密操作のバランスが取りづらく難儀しています。
DPI調整のためのキーコードも使っていますが、「精密操作が必要な今この瞬間だけDPIを1/3位にしたい」という場合には少々使いづらいと感じます。
DPI設定を複数保持してトグル操作かモメンタム操作で切り替えられるようになるか、あるいは本来の設定値の1/3程度にトグル操作かモメンタム操作で切り替えられるようになればありがたいと感じますが、いかがでしょうか。

以上、ご検討いただければ幸いです。

**Comment by Yowkees** [comment]
ご提案ありがとうございます。

ぱっと思いつくのは、特定のレイヤーに切り替え中にDPIを下げる処理をする方法で、ファームウェアを書き換えるのが早いかと思います。設定で切り替えられるようにするには少々方法を検討する必要があると思います。
本ファームウェアに導入できるかは検討したいと思います。

以上、よろしくお願いいたします。

**Reply by garbx86** [reply]
ご回答頂き、大変ありがとうございます。

ファームウェアに関してはあまり知識がありませんが、レイヤー切替による処理について勉強してみます。

今後ともどうかよろしくお願いいたします。

**Comment by ryicoh** [comment]
> 、特定のレイヤーに切り替え中にDPIを下げる処理をする方法で

動いているか怪しいですが、↑をやってみました。参考までに。

https://github.com/ryicoh/keyball/commit/adf4a05b6e840ef6ddb81123ed8d2786d6f93399
